devtools/maven/src/main/java/io/quarkus/maven/BuildMojo.java

```diff
@@ -173,13 +174,21 @@ public void execute() throws MojoExecutionException {
                     projectArtifact.getVersion());
             appArtifact.setPaths(PathsCollection.of(projectArtifact.getFile().toPath()));
 
-            CuratedApplication curatedApplication = QuarkusBootstrap.builder()
+            QuarkusBootstrap.Builder builder = QuarkusBootstrap.builder()
                     .setAppArtifact(appArtifact)
                     .setMavenArtifactResolver(resolver)
                     .setBaseClassLoader(BuildMojo.class.getClassLoader())
                     .setBuildSystemProperties(realProperties)
+                    .setLocalProjectDiscovery(false)
+                    .setProjectRoot(project.getBasedir().toPath())
                     .setBaseName(finalName)
-                    .setTargetDirectory(buildDir.toPath())
+                    .setTargetDirectory(buildDir.toPath());
+
+            for (MavenProject project : project.getCollectedProjects()) {
+                builder.addLocalArtifact(new AppArtifactKey(project.getGroupId(), project.getArtifactId(), null, "jar"));
```

Comment:
Instead of assuming it's a jar you should probably use `project.getArtifact().getArtifactHandler().getExtension()`.